ProcessLib/SmallDeformationWithLIE/LocalAssembler/SmallDeformationLocalAssemblerFracture-impl.h

```diff
@@ -0,0 +1,174 @@
+        ip_data._sigma_prev.resize(DisplacementDim);
+        ip_data._C.resize(DisplacementDim, DisplacementDim);
+        ip_data._aperture0 = (*frac_prop->aperture0)(0, x_position)[0];
+        ip_data._aperture_prev = ip_data._aperture0;
+
+        _secondary_data.N[ip] = sm.N;
+    }
+}
+
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          int DisplacementDim>
+void SmallDeformationLocalAssemblerFracture<ShapeFunction, IntegrationMethod,
+                                    DisplacementDim>::
+assembleWithJacobian(
+    double const t,
+    Eigen::VectorXd const& local_u,
+    Eigen::VectorXd& local_b,
+    Eigen::MatrixXd& local_J)
+{
+    auto const& nodal_jump = local_u;
+
+    FractureProperty const& frac_prop = *_process_data._fracture_property;
+    auto const& R = frac_prop.R;
+
+    int const index_normal = DisplacementDim - 1;
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    SpatialPosition x_position;
+    x_position.setElementID(_element.getID());
+
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        x_position.setIntegrationPoint(ip);
+        auto const& wp = _integration_method.getWeightedPoint(ip);
+
+        auto& ip_data = _ip_data[ip];
+        auto const& detJ = ip_data._detJ;
+        auto const& integralMeasure = ip_data._integralMeasure;
+        auto const& H = ip_data._h_matrices;
+        auto& mat = ip_data._fracture_material;
+        auto& sigma = ip_data._sigma;
+        auto const& sigma_prev = ip_data._sigma_prev;
+        auto& w = ip_data._w;
+        auto const& w_prev = ip_data._w_prev;
+        auto& C = ip_data._C;
+
+        // displacement jumps
+        w.noalias() = R * H * nodal_jump;
+
+        // total aperture
+        ip_data._aperture = ip_data._aperture0 + w[index_normal];
+
+        // local C, local stress
+        mat.computeConstitutiveRelation(
+                    t, x_position,
+                    w_prev, w,
+                    sigma_prev, sigma, C);
+
+        // r_[u] += H^T*Stress
+        local_b.noalias() -= H.transpose() * R.transpose() * sigma * detJ * wp.getWeight() * integralMeasure;
+
+        // J_[u][u] += H^T*C*H
+        local_J.noalias() += H.transpose() * R.transpose() * C * R * H * detJ * wp.getWeight() * integralMeasure;
+    }
+
+}
+
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          int DisplacementDim>
+void SmallDeformationLocalAssemblerFracture<ShapeFunction, IntegrationMethod,
+                                    DisplacementDim>::
+postTimestepConcrete(std::vector<double> const& /*local_x*/)
+{
+    double ele_b = 0;
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        ele_b += _ip_data[ip]._aperture;
+    }
+    ele_b /= n_integration_points;
+    (*_process_data._mesh_prop_b)[_element.getID()] = ele_b;
```

Comment:
Do you use this for auxiliary output?
For now it's OK. But our output functionality clearly has to be improved in this regard (Actually, I wanted to do this already several months ago...)